R/app_server.R

```diff
@@ -223,6 +223,38 @@ app_server <- function(input, output, session) {
     get_comm_data(selected_pkg$name())
   })
   
+  loaded2_db <- eventReactive(selected_pkg$name(), {
+    req(selected_pkg$name())
+    req(selected_pkg$name() != "-")
+    
+    dbSelect("SELECT name, version, score FROM package")
+  })
+  
+  # Get Package Dependency metrics.
+  dep_metrics  <- reactiveVal()
+  
+  pkgref <- eventReactive(selected_pkg$name(), {
+    req(selected_pkg$name())
+    req(selected_pkg$name() != "-")
+
+    get_assess_blob(selected_pkg$name())
+  })
+  
+  observeEvent(pkgref(), {
+    req(pkgref())
+    tryCatch(
+      expr = {
+        dep_metrics(pkgref()$dependencies[[1]] %>% dplyr::as_tibble())
+      },
+      error = function(e) {
+        msg <- paste("Detailed dependency information is not available for package", selected_pkg$name())
+        rlang::warn(msg)
+        rlang::warn(paste("info:", e))
+        dep_metrics(dplyr::tibble(package = character(0), type = character(0), name = character(0)))
+      }
+    )
+  })
+
```

Comment:
This isn't going to work for the database view tab. Instead of grabbing the dependency information for each package, this pastes the dependency information for the selected package on the sidebar and pastes it into each report.

In fact if you go straight to the database view tab and never select a package, the app will crash if you try to download a report.

Reply:
OK I got this one moved